wp-content/themes/pub/wporg-learn-2020/template-parts/component-breadcrumbs.php

```diff
@@ -14,20 +14,35 @@
 	),
 );
 
-// Get information about the post title.
-$cpt_object = get_post_type_object( get_post_type( get_queried_object() ) );
-
-if ( 'lesson-plan' === get_post_type() ) {
+if ( is_post_type_archive() ) {
 	array_push( $crumbs, array(
-		'label' => ucfirst( $cpt_object->labels->name ),
+		'label' => post_type_archive_title( '', false ),
+		'url' => '',
+	));
+} elseif ( is_singular( 'lesson-plan' )
+	|| is_singular( 'wporg_workshop' )
+	|| is_singular( 'course' ) ) {
```

Comment:
Explicitly check every post type we have for now.
Not sure if it's a good idea. Open to discussion.

Reply:
That's a good question! I think we probably don't want the extra work you mentioned, since there's a chance we won't remember to come back here and update it when we add a new post type. I think the logic will work for all post types that are visible on the front end.

For the `page` case, I think it'd be good to have it documented somehow, otherwise our future selves won't know what specifically is being affected by that condition.

So I'd probably collapse single posts and pages like the following. It reduces the duplicate code, and feels more "self documenting". 

```
if ( is singular ) {
	if ( is single ) {
		get post type object
		push archive link onto array
	}

	push title onto array
}
```

Pages aren't explicitly mentioned, but it's implied because pages are the only thing that match `is_singular()` but not `is_single()`. A comment could explicitly mention them too, though, if you find that helpful.

Reply:
Thanks @iandunn, the explanation about extra work needed and `page` case makes a good point!

I've revised the logic based on your suggestion, it's indeed clearer and more explicit.

I'm not going to add extra comments here for the moment as it seems to me that it might bring extra confusion by adding some short and incomplete descriptions. I think the explanation in the WP API document would be clear enough to understand.

Reply:
Here's the new [commit](https://github.com/WordPress/learn/pull/658/commits/e5dd69c2c7f495a50c9fcd35e59d780db48a8738) and the new [screencast](https://d.pr/v/yPBmOL) of the revised logic - everything still works as expected.

Reply:
Thanks @iandunn, the explanation about extra work needed and page case makes a good point!

I've revised the logic based on your suggestion, and I'm not going to add extra comments here for the moment as it seems to me that it might bring extra confusion by adding some short and incomplete descriptions. I think the explanation in the WP API document would be clear enough to understand.

Here's the new [commit](https://github.com/WordPress/learn/pull/658/commits/e5dd69c2c7f495a50c9fcd35e59d780db48a8738) and the new [screencast](https://d.pr/v/yPBmOL) of the revised logic - everything still works as expected.

Reply:
Looks good!

Does the final `else` condition still do anything, now that pages are covered by the `is_singular()` case?

If not, I'd lean towards removing it, to avoid clutter and confusion. If if something we didn't think of does trigger it, `get_the_title()` might not work anyway.

Reply:
> If something we didn't think of does trigger it, then get_the_title() might not work anyway.

Ah, you're right. Let me remove it. And then I'll merge the PR.
Thanks!